src/operator/tensor/matrix_op-inl.h

```diff
@@ -11,6 +11,14 @@
 #include "../mshadow_op.h"
 #include "../elemwise_op_common.h"
 
+namespace dmlc {
+
+// Add a few patches to support signed Shape in dmlc/parameter.
+DMLC_DECLARE_TYPE_NAME(nnvm::Tuple<int>, "Shape(tuple)");
```

Comment:
I feel that it's not so good to add it in the matrix_op-inl.h since the Tuple<int> will be used in other files.